src/itwinai/torch/inference.py

```diff
@@ -144,11 +235,49 @@ def execute(
                 else:
                     pre = pre.to_dense().tolist()
                 all_predictions[idx] = pre
+
+        if self.logger:
+            self.logger.destroy_logger_context()
+
+        self.strategy.clean_up()
+
         return all_predictions
 
-    @abc.abstractmethod
+    def log(
+        self,
+        item: Union[Any, List[Any]],
+        identifier: Union[str, List[str]],
+        kind: str = 'metric',
+        step: Optional[int] = None,
+        batch_idx: Optional[int] = None,
```

Comment:
When doing type hints, we have decided to use "|" instead of "Union" and "| None" instead of "Optional" in codebase.